addons/mass_mailing/controllers/main.py

```diff
@@ -167,23 +169,16 @@ def view(self, mailing_id, email=None, res_id=None, token=""):
             if not self._valid_unsubscribe_token(mailing_id, res_id, email, str(token)) and not request.env.user.has_group('mass_mailing.group_mass_mailing_user'):
                 raise exceptions.AccessDenied()
 
-            res = mailing.convert_links()
-            base_url = mailing.get_base_url().rstrip('/')
-            urls_to_replace = [
-                (base_url + '/unsubscribe_from_list', mailing._get_unsubscribe_url(email, res_id)),
-                (base_url + '/view', mailing._get_view_url(email, res_id))
-            ]
-            for url_to_replace, new_url in urls_to_replace:
-                if url_to_replace in res[mailing_id]:
-                    res[mailing_id] = res[mailing_id].replace(url_to_replace, new_url if new_url else '#')
-
-            res[mailing_id] = res[mailing_id].replace(
-                'class="o_snippet_view_in_browser"',
-                'class="o_snippet_view_in_browser" style="display: none;"'
-            )
+            html = str(request.env['mail.render.mixin'].sudo()._render_template_qweb(
```

Comment:
This "str" is strange ^^

Reply:
Actually, this was for injecting the “display:none” for the “View online” link which makes no sense when viewing it online. When not converting to a string, it remains a markupsafe.Markup and the replace method avoid injection of untrusted code like this. 
But I have found another solution to hide the “View online” so it is not needed anymore: at the page render, the style “.o_snippet_view_in_browser { display: none; }” is added in the header.